catalog/task/collect-extra-data/0.2/collect-extra-data.yaml

```diff
@@ -0,0 +1,101 @@
+  params:
+    - name: release
+      type: string
+      description: The namespaced name of the Release
+    - name: releasePlan
+      type: string
+      description: The namespaced name of the ReleasePlan
+    - name: releasePlanAdmission
+      type: string
+      description: The namespaced name of the ReleasePlanAdmission
+    - name: snapshot
+      type: string
+      description: The namespaced name of the Snapshot
+    - name: saveRelease
+      type: string
+      description: Save the Release CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlan
+      type: string
+      description: Save the ReleasePlan CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlanAdmission
+      type: string
+      description: Save the ReleasePlanAdmission CR as a json in the data workspace
+      default: true
+    - name: saveSnapshot
+      type: string
+      description: Save the Snapshot CR spec as a json in the data workspace
+      default: true
+  workspaces:
+    - name: data
+      description: Workspace to save the CR jsons to
+  results:
+    - name: extraData
+      description: |
+        Result of merging the `extraData` field of all the release resources
+  steps:
+    - name: collect-extra-data
+      image: quay.io/hacbs-release/release-utils:f486f350df43974706fc484a13bfeefc2c8fa95a
+      env:
+        - name: "RELEASE"
+          value: '$(params.release)'
+        - name: "RELEASE_PLAN"
+          value: '$(params.releasePlan)'
+        - name: "RELEASE_PLAN_ADMISSION"
+          value: '$(params.releasePlanAdmission)'
+        - name: "SNAPSHOT"
+          value: '$(params.snapshot)'
+      script: |
+        #!/usr/bin/env sh
+
+        if ! release_result=$(get-resource "release" "${RELEASE}" "{.spec.extraData}"); then
+          exit 1
+        fi
+
+        if ! release_plan_result=$(get-resource "releaseplan" "${RELEASE_PLAN}" "{.spec.extraData}"); then
+          exit 1
+        fi
+
+        if ! release_plan_admission_result=$(get-resource "releaseplanadmission" "${RELEASE_PLAN_ADMISSION}" \
+            "{.spec.extraData}"); then
+          exit 1
+        fi
+
+        if [ $(params.saveRelease) = true ] ; then
+            get-resource "release" "${RELEASE}" > "$(workspaces.data.path)/release.json"
+        fi
+
+        if [ $(params.saveReleasePlan) = true ] ; then
+            get-resource "releaseplan" "${RELEASE_PLAN}" > "$(workspaces.data.path)/release_plan.json"
+        fi
+
+        if [ $(params.saveReleasePlanAdmission) = true ] ; then
+            get-resource "releaseplanadmission" "${RELEASE_PLAN_ADMISSION}" \
+              > "$(workspaces.data.path)/release_plan_admission.json"
+        fi
+
+        if [ $(params.saveSnapshot) = true ] ; then
+            get-resource "snapshot" "${SNAPSHOT}" "{.spec}" > "$(workspaces.data.path)/snapshot_spec.json"
```

Comment:
Maybe the parameter should be called `saveSnapshotSpec` then? WDYT?

Reply:
David actually wants me to remove all the save* params

Reply:
Yes, this was opened as I was working on 2290. So all the pipelines will run this task first to put the snapshot in a workspace and then the other tasks will pull it from there instead of as a parameter